dashboard/app/models/pd/application/teacher_application_base.rb

```diff
@@ -65,6 +64,20 @@ def update_user_school_info!
       end
     end
 
+    def update_scholarship_status(scholarship_status)
+      scholarship_info = Pd::ScholarshipInfo.find_by(user: user, application_year: application_year) || Pd::ScholarshipInfo.new(user: user)
+      scholarship_info.update(scholarship_status: scholarship_status, pd_application_id: id)
+    end
+
+    # also adds the pd_application_id to the ScholarshipInfo object
```

Comment:
This method having a side effect feels like a code smell to me. A (probably better) alternative would be to look for an application (and a summer workshop enrollment) to associate whenever we create or update a ScholarshipInfo object. Actually though, I'm not sure how useful the association with application and enrollment are, since the unique identifier of a ScholarshipInfo is user and year, so that is the most reliable way to find the one we want. Thoughts?

Reply:
Hmm... that's a good point.  I'd definitely remove the side-effect here.  If there's no need to ever associate a scholarship back to a particular application or enrollment, maybe we should just remove those columns? (Sorry I didn't think through this earlier)

Reply:
Associating scholarships back to applications or enrollments sounds hypothetically useful, but not a use case we have at the moment. Right now we only interact with ScholarshipInfo when we are updating the scholarship info for a given application or enrollment. We never start with the scholarship info and need to find the application or enrollment.